node/src/service.rs

```diff
@@ -665,6 +646,31 @@ async fn start_node_impl(
         );
     }
 
+    // If node is running as a Storage Provider, start building the StorageHubHandler using the StorageHubBuilder.
+    let mut sh_builder = None;
+    if provider_options.is_some() {
```

Comment:
We could extract everything in this if statement into a function, since it is the same for both the node and dev modes.